src/CommunityCommons/javasource/communitycommons/Misc.java

```diff
@@ -265,7 +265,7 @@ public static IContext getContextFor(IContext context, String username, boolean
         }
 
         // Session does not have a user when it's a scheduled event.
-        if (context.getSession().getUser() != null && username.equals(context.getSession().getUser().getName())) {
+        if (username.equals(context.getSession().getUserName())) {
```

Comment:
The comment above this line does no longer make sense with your implementation.

Reply:
I updated the comment.